File: src/Helpers/BrokerProvider.java
package Helpers;


import Models.Broker;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class BrokerProvider {

    public static List<Broker> fetchBrokers() {
        Scanner input = null;
        try {
            input = new Scanner(new File("/home/p3160253/IdeaProjects/distributed/src/Data/brokers.txt"));
        } catch (FileNotFoundException e) {
            e.printStackTrace();
        }
        input.useDelimiter("-\n");
        List<Broker> brokers = new ArrayList<>();
        while (input.hasNextLine()) {
            String data = input.nextLine();
            String ip = data.split(":")[0];
            int port = Integer.parseInt(data.split(":")[1]);
            Broker broker = new Broker(ip, port);
            brokers.add(broker);
        }
        return brokers;
    }
}

File: src/helpers/BrokerProvider.kt
package helpers

import models.Broker

import java.io.File
import java.io.FileNotFoundException
import java.util.ArrayList
import java.util.Scanner

object BrokerProvider {
    fun fetchBrokers(): List<Broker> {
        var input: Scanner? = null
        try {
            input = Scanner(File("/Users/jimfilippou/Projects/distributed/src/data/brokers.txt"))
        } catch (e: FileNotFoundException) {
            e.printStackTrace()
        }
        input!!.useDelimiter("-\n")
        val brokers = ArrayList<Broker>()
        while (input.hasNextLine()) {
            val data = input.nextLine()
            val ip = data.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[0]
            val port = Integer.parseInt(data.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()[1])
            val broker = Broker(ip, port)
            brokers.add(broker)
        }
        return brokers
    }
}